src/main/java/io/cdap/plugin/salesforce/plugin/source/streaming/SalesforcePushTopicListener.java

```diff
@@ -129,20 +129,95 @@ protected void customize(Request exchange) {
 
     // Now set up the Bayeux client itself
     BayeuxClient client = new BayeuxClient(oAuthInfo.getInstanceURL() + DEFAULT_PUSH_ENDPOINT, transport);
-    client.handshake();
-
     return client;
   }
 
-  private void waitForHandshake(BayeuxClient client,
-                                long timeoutInMilliseconds, long intervalInMilliseconds) {
+  public void createSalesforceListener() {
+    try {
+      bayeuxClient = getClient(credentials);
+    } catch (Exception exception) {
+      LOG.error("Exception while creating bayeux client ", exception);
+    }
+    bayeuxClient.getChannel(Channel.META_HANDSHAKE).addListener
+      ((ClientSessionChannel.MessageListener) (channel, message) -> {
+
+        boolean success = message.isSuccessful();
+        if (!success) {
+          String error = (String) message.get("error");
+          if (error != null) {
+            LOG.error("Error during HANDSHAKE: {}", error);
+          }
+
+          Exception exception = (Exception) message.get("exception");
```

Comment:
How do we know that message contains `error`, `exception` keys? Is it documented somewhere?
Also, it would be safer to check if  message.get("exception") instanceof Exception before casting